app/src/ui/lib/list/list.tsx

```diff
@@ -241,33 +239,25 @@ interface IListProps {
   /** The row that should be scrolled to when the list is rendered. */
   readonly scrollToRow?: number
 
-  /** Whether or not selection should follow pointer device */
-  readonly selectOnHover?: boolean
-
   /** Type of elements that can be inserted in the list via drag & drop. Optional. */
   readonly insertionDragType?: DragType
 
-  /**
-   * Whether or not to explicitly move focus to a row if it was selected
-   * by hovering (has no effect if selectOnHover is not set). Defaults to
-   * true if not defined.
-   */
-  readonly focusOnHover?: boolean
-
   /**
    * The number of pixels from the top of the list indicating
    * where to scroll do on rendering of the list.
    */
   readonly setScrollTop?: number
 
   /**
-   * Optional callback for providing an aria label for screen readers for each
-   * row.
-   *
-   * Note: you may need to apply an aria-hidden attribute to any child text
-   * elements for this to take precedence.
+   * Optional callback for providing ane or more aria attributes for a given row
    */
-  readonly getRowAriaLabel?: (row: number) => string | undefined
+  readonly getRowAccessibilityProps?: (row: number) => AccessibilityProps
+
+  /** The aria-labelledby attribute for the list component. */
+  readonly 'aria-labelledby'?: string
+
+  /** The aria-label attribute for the list component. */
+  readonly 'aria-label'?: string
```

Comment:
Seems like this could be a use case for the new AccessibilityProps? 

I am thinking that we should make AccessibilityProps be the preferred way to add aria attributes so we have consistent approach. With this PR, we now have three approaches `ariaLabel` (camel casing props), `aria-label` (quoting props), and using the `AccessibilityProps`.  Accessibility props does give us an easy range of inclusion of any aria-label we want... only downside I see is that it almost indicates that any of those aria-props are valid, but some like `aria-checked` only make since things like checkboxes and radio buttons. :/ 

If we don't use the `AccessibilityProps`, I prefer using the camel causing on react components as it makes it clear that it is a react prop not an aria attribute implementation which you can add to react component whether or not there is a prop to receive it.

Reply:
I looked at polishing this up to a point where we could switch to using accessibility props all over but there are nuances here around ensuring we don't accidentally pick props not applicable to the component and decided that this isn't worth spending time on right now.